x/model/keeper/msg_server_model_version.go

```diff
@@ -135,3 +135,55 @@ func (k msgServer) UpdateModelVersion(goCtx context.Context, msg *types.MsgUpdat
 
 	return &types.MsgUpdateModelVersionResponse{}, nil
 }
+
+func (k msgServer) DeleteModelVersion(goCtx context.Context, msg *types.MsgDeleteModelVersion) (*types.MsgDeleteModelVersionResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check signer has enough rights to delete model version
+	signerAddr, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid Address: (%s)", err)
+	}
+	if err := checkModelRights(ctx, k.Keeper, signerAddr, msg.Vid, "MsgDeleteModelVersion"); err != nil {
+		return nil, err
+	}
+
+	// check if model version exists
+	modelVersion, isFound := k.GetModelVersion(
+		ctx,
+		msg.Vid,
+		msg.Pid,
+		msg.SoftwareVersion,
+	)
+
+	if !isFound {
+		return nil, types.NewErrModelVersionDoesNotExist(msg.Vid, msg.Pid, msg.SoftwareVersion)
+	}
+
+	if msg.Creator != modelVersion.Creator {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized, "MsgDeleteModelVersion transaction should be "+
+			"signed by a modelVersion %d %d %d creator", msg.Vid, msg.Pid, msg.SoftwareVersion)
+	}
+
+	isCertified := k.IsModelCertified(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion)
+	if isCertified {
+		return nil, types.NewErrModelVersionCertified(msg.Vid, msg.Pid, modelVersion.SoftwareVersion)
+	}
+
+	// store updated model version
+	k.RemoveModelVersion(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion)
+
+	return &types.MsgDeleteModelVersionResponse{}, nil
+}
+
+func (k msgServer) IsModelCertified(ctx sdk.Context, vid int32, pid int32, softwareVersion uint32) bool {
+	certificationTypes := []string{"zigbee", "matter"}
```

Comment:
Please use existing `CertificationTypesList` constant, not these hard-coded values (actually this is not all supported certification types, there are 4 values in CertificationTypesList).

Reply:
Fixed